Feature: ResultCard component

chancestrickland opened the pull request:
This PR creates a new `ResultCard` component as well as more generic composable `Card` components. Closes #22.

**IMPORTANT:** There are edits to the homepage that should not be merged. These are intended to make it quick to review these components locally.

Component API:

```js
<ResultCard
  // assumes these come from a list somewhere, most likely
  businessCategory="foodAndBeverage"
  businessName="Dad's Donuts"
  businessDescription="Blah blah blaaaah"
  businessLocation="Atlanta, GA"
  businessUrl="https://dobusiness.com"
  donationUrl="https://donatemoney.com"
  // overrides the default fallback image
  imageSrc="https://imagesrc.jpg"
  imageAlt="cool image you got there"
/>
```

### Screenshots / video of changes

![image](https://user-images.githubusercontent.com/3082153/83926529-e7fab680-a73e-11ea-8fae-f4af9cbc4fb5.png)

## Steps to test

Run locally and check out the homepage.

chancestrickland commented:
OH and BTW, I added eslint dependencies because...well, there was no linting and I was making lots of mistakes. I can put this in another PR if it's too cluttery here, or if someone else is handling this elsewhere. Just let me know.

magnificode commented:
@chancestrickland this is looking **solid** - Is there any chance we can incorporate the work @kkemple did here https://github.com/Rebuild-Black-Business/RBB-Website/pull/35 - Just want to make sure we get proper schema markup baked into the ResultCard component for businesses.

Also very much appreciate you adding that linting. Need to have that in sooner rather than later for suuure.

racedale commented:
@chancestrickland some linting was merged to master today, make sure to merge master in!

ChrisBrownie55 commented on src/components/ResultCard.js:
```diff
@@ -0,0 +1,158 @@
+      children,
+      imageSrc,
+      imageAlt,
+      businessCategory,
+      businessName,
+      businessDescription,
+      businessLocation,
```
It seems a tad redundant to have `business` in front of each word. Some options to make the code a little less verbose could be an **object with all of the business stuff** in it (i.e. `business.category`, `business.name`) and that could _also apply to things like the image_ (i.e. `image.src`, `image.alt`).

Or another option could just be having the unique names be without the first part (i.e. `src`, `alt`, `category`, `name`) but that could be confusing.

chancestrickland replied:
Prefer to stay away from object literal props as a general rule if I can, but happy to drop the business prefix.

chancestrickland commented:
@magnificode I incorporated some of @kkemple's patterns, though it seems like we still need to flesh out the categories a bit. Left some todos in place but let me know those are going to be handled. I imagine we'll pull something from Airtable in whatever context these components are used rather than the component itself, so we can probably revisit this after merging. 

@racedale I updated and merged with master.

chancestrickland commented on package.json:
```diff
@@ -47,16 +47,25 @@
     "eslint-config-react-app": "^5.2.1",
     "eslint-plugin-chai-friendly": "^0.6.0",
     "eslint-plugin-cypress": "^2.11.1",
+    "eslint-plugin-flowtype": "4.x",
```
Just a note, all the new deps here are all peer deps for `eslint-config-react-app`.

magnificode commented:
> @magnificode I incorporated some of @kkemple's patterns, though it seems like we still need to flesh out the categories a bit. Left some todos in place but let me know those are going to be handled. I imagine we'll pull something from Airtable in whatever context these components are used rather than the component itself, so we can probably revisit this after merging.
> 
> @racedale I updated and merged with master.

Yeah no worries at all, appreciate you keeping things flexible there and adding to-dos. This will continue to be a bit fluid as designs start rolling in. Very much appreciate the work here @chancestrickland !

magnificode commented:
@chancestrickland - feel free to remove the examples on the home page for now! If you wanted to turn the snippet you have in your original PR message into commented documentation in the Component file itself that would be awesome! Just to get a little usage example in the file itself. 

Ping me when you've got that handled and I'll merge this in. Thank you again for the great work!

jakeleboeuf commented on src/components/Button.js:
```diff
@@ -38,12 +39,20 @@ function getButtonStyles(theme, variant) {
+});
+
+Button.displayName = 'Button';
+Button.propTypes = {
+  variant: PropTypes.oneOf(['cta', 'primary', 'secondary']),
+};
```
I've been in Typescript world for so long I forgot about PropTypes 😍

chancestrickland replied:
OMG SAME

chancestrickland commented:
@magnificode Removed the homepage stuff, added that snippet in a JSDoc block. Holler if you need anything else to 🚢
